server/src/domain/asset/asset.service.ts

```diff
@@ -141,4 +149,116 @@ export class AssetService {
 
     throw new BadRequestException('assetIds, albumId, or userId is required');
   }
+
+  async handleRefreshAsset(job: ILibraryJob) {
+    console.log('Refreshing file ' + job.assetPath);
+    // TODO: Determine file type from extension only
+    const mimeType = mime.lookup(job.assetPath);
+    if (!mimeType) {
+      throw Error(`Cannot determine mime type of asset: ${job.assetPath}`);
+    }
+    if (!isSupportedFileType(mimeType)) {
+      throw new BadRequestException(`Unsupported file type ${mimeType}`);
+    }
+
+    const existingAssetEntity = await this.assetRepository.getByLibraryIdAndOriginalPath(job.libraryId, job.assetPath);
+
+    let stats: fs.Stats;
+    try {
+      stats = await fs.promises.stat(job.assetPath);
+    } catch (error) {
+      // Can't access file, probably offline
+      if (job.emptyTrash && existingAssetEntity) {
+        // Remove asset from database
+        await this.assetRepository.remove(existingAssetEntity);
+        return true;
+      } else if (existingAssetEntity) {
+        // Mark asset as offline
+        existingAssetEntity.isOffline = true;
+        await this.assetRepository.save(existingAssetEntity);
+        return true;
+      } else {
+        throw new BadRequestException(error, "Can't access file");
+      }
+    }
+
+    if (existingAssetEntity && stats.mtime == existingAssetEntity.fileModifiedAt && !job.forceRefresh) {
+      // File last modified time matches database entry
+      // Unless we're forcing a refresh, exit here
+      return true;
+    }
```

Comment:
Perhaps there should be a way to reset `isOffline` to `false` is the asset can be accessed again. I'm thinking of cases where a user is using rclone as the backend and there was a temporary connectivity issue.

Reply:
The behavior you describe is the intended behavior. I seem to have forgotten the code to set offline to false, will fix